Testsuite: all Tomcat ports needs to be offset, not just 8080

Given these use host networking, the ajp connector port needs to be offset as well.

```
10-Aug-2023 11:20:54.790 SEVERE [main] org.apache.catalina.core.StandardService.initInternal Failed to initialize connector [Connector[AJP/1.3-8009]]
	org.apache.catalina.LifecycleException: Protocol handler initialization failed
		at org.apache.catalina.connector.Connector.initInternal(Connector.java:1075)
		at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:136)
		at org.apache.catalina.core.StandardService.initInternal(StandardService.java:559)
		at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:136)
		at org.apache.catalina.core.StandardServer.initInternal(StandardServer.java:854)
		at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:136)
		at org.apache.catalina.startup.Catalina.load(Catalina.java:646)
		at org.apache.catalina.startup.Catalina.load(Catalina.java:669)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:568)
		at org.apache.catalina.startup.Bootstrap.load(Bootstrap.java:307)
		at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:477)
	Caused by: java.net.BindException: Address already in use
		at java.base/sun.nio.ch.Net.bind0(Native Method)
		at java.base/sun.nio.ch.Net.bind(Net.java:555)
		at java.base/sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:337)
		at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:294)
		at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:89)
		at org.apache.tomcat.util.net.NioEndpoint.initServerSocket(NioEndpoint.java:226)
		at org.apache.tomcat.util.net.NioEndpoint.bind(NioEndpoint.java:202)
		at org.apache.tomcat.util.net.AbstractEndpoint.bindWithCleanup(AbstractEndpoint.java:1318)
		at org.apache.tomcat.util.net.AbstractEndpoint.init(AbstractEndpoint.java:1331)
		at org.apache.tomcat.util.net.AbstractJsseEndpoint.init(AbstractJsseEndpoint.java:231)
		at org.apache.coyote.AbstractProtocol.init(AbstractProtocol.java:693)
		at org.apache.catalina.connector.Connector.initInternal(Connector.java:1073)
		... 13 more
```

--- a/test/includes/common.sh
+++ b/test/includes/common.sh
@@ -114,20 +114,30 @@ tomcat_create() {
 # Start tomcat$1 container on 127.0.0.$2
 # or 127.0.0.$1 if $2 is not given
 # arguments:
-#       $1 tomcat number
+#       $1 tomcat number (required)
 #       $2 tomcat's last byte of IPv4 address (if 0 or omitted, equals to $1)
-#       $3 tomcat port (if omitted it's 8080)
+#       $3 tomcat port          (if omitted it's 8080 + $1 - 1)
+#       $4 tomcat ajp port      (if omitted it's 8900 + $1 - 1)
+#       $5 tomcat shutdown port (if omitted it's 8005 + $1 - 1)
 tomcat_start() {
+    if [ -z "$1" ]; then
+        echo "tomcat_start called without arguments"
+        exit 1
+    fi
     ADDR="127.0.0.$1"
     if [ ${2:-0} -ne 0 ]; then
         ADDR="127.0.0.$2"
     fi
 
+    local portdef=$(expr 8080 + $1 - 1)
+    local ajpdef=$(expr 8900 + $1 - 1)
+    local shutdef=$(expr 8005 + $1 - 1)
+
     echo "Starting tomcat$1 on $ADDR"
-    nohup docker run --network=host -e tomcat_ajp_port=8010 \
+    nohup docker run --network=host -e tomcat_ajp_port=${4:-$ajpdef} \
                                     -e tomcat_address=$ADDR \
-                                    -e tomcat_port=${3:-8080} \
-                                    -e tomcat_shutdown_port=8005 \
+                                    -e tomcat_port=${3:-$portdef} \
+                                    -e tomcat_shutdown_port=${5:-$shutdef} \
                                     -e jvm_route=tomcat$1 \
                                 --name tomcat$1 ${IMG} &
     ps -q $! > /dev/null
@@ -250,6 +260,10 @@ tomcat_start_webapp() {
 }
 
 # Send a shutdown packet to a tomcat$1 container
+# arguments:
+#     $1 tomcat number
+#     $2 the last segment of IPv4 addr ($1 by default)
+#     $3 the shutdown port (8005 + $1 - 1 by default)
 tomcat_shutdown() {
     ADDR="127.0.0.$1"
     if [ $2 -ne 0 ]; then
@@ -257,7 +271,7 @@ tomcat_shutdown() {
     fi
 
     echo "shutting down tomcat$1 with address: $ADDR"
-    echo "SHUTDOWN" | nc $ADDR 8005
+    echo "SHUTDOWN" | nc $ADDR ${3:-$(expr 8005 + $1 - 1)}
 }
 
 # Remove the docker image tomcat$1

--- a/test/tomcat/server.xml
+++ b/test/tomcat/server.xml
@@ -19,7 +19,7 @@
      define subcomponents such as "Valves" at this level.
      Documentation at /docs/config/server.html
  -->
-<Server port="8005" shutdown="SHUTDOWN" address="127.0.0.1">
+<Server port="shutdown_port" shutdown="SHUTDOWN" address="127.0.0.1">
   <!-- <Listener className="org.jboss.modcluster.container.tomcat.ModClusterListener" advertise="true" stickySession="true" stickySessionForce="false" stickySessionRemove="true" /> -->
   <Listener className="org.jboss.modcluster.container.catalina.standalone.ModClusterListener" connectorPort="changeMe" advertise="false" stickySession="true" stickySessionForce="false" stickySessionRemove="true" proxyList="proxyaddress:proxyport"/>
   <Listener className="org.apache.catalina.startup.VersionLoggerListener" />
@@ -68,7 +68,7 @@
          APR (HTTP/AJP) Connector: /docs/apr.html
          Define a non-SSL/TLS HTTP/1.1 Connector on port 8080
     -->
-    <Connector port="8080" protocol="HTTP/1.1"
+    <Connector port="tomcat_port" protocol="HTTP/1.1"
                address="127.0.0.1"
                connectionTimeout="20000"
                redirectPort="8443" />
@@ -119,7 +119,7 @@
     
     <Connector protocol="AJP/1.3"
                address="127.0.0.1"
-               port="8009"
+               port="ajp_port"
                secretRequired="false"
                redirectPort="8443" />
    

--- a/test/tomcat/start.sh
+++ b/test/tomcat/start.sh
@@ -16,15 +16,15 @@ fi
 
 
 if [ ! -z ${tomcat_port} ]; then
-  sed -i "s/8080/${tomcat_port}/" ./conf/server.xml
+  sed -i "s/tomcat_port/${tomcat_port}/" ./conf/server.xml
 fi
 
 if [ ! -z ${tomcat_shutdown_port} ]; then
-  sed -i "s/8005/${tomcat_shutdown_port}/" ./conf/server.xml
+  sed -i "s/shutdown_port/${tomcat_shutdown_port}/" ./conf/server.xml
 fi
 
 if [ ! -z ${tomcat_ajp_port} ]; then
-  sed -i "s/8009/${tomcat_ajp_port}/" ./conf/server.xml
+  sed -i "s/ajp_port/${tomcat_ajp_port}/" ./conf/server.xml
 fi
 if [ ! -z ${tomcat_address} ]; then
   sed -i "s/127.0.0.1/${tomcat_address}/" ./conf/server.xml
@@ -47,7 +47,7 @@ fi
 sed -i "s/proxyport/${cluster_port}/" ./conf/server.xml
 sed -i "s/proxyaddress/127.0.0.1/" ./conf/server.xml
 
-echo "jvm_route: ${jvm_route} and tomcat_port: ${tomcat_port} and tomcat_address: ${tomcat_address}"
+echo "jvm_route: ${jvm_route:-default} tomcat_port: ${tomcat_port:-default} tomcat_address: ${tomcat_address:-default} tomcat_ajp_port: ${tomcat_ajp_port:-default} tomcat_shutdown_port: ${tomcat_shutdown_port:-default}"
 if [ ! -z ${jvm_route} ]; then
   sed -i "/<Engine name=\"Catalina\"/c\<Engine name=\"Catalina\" defaultHost=\"localhost\" jvmRoute=\"${jvm_route}\">" ./conf/server.xml
 fi

